### PR TITLE
[qontract-cli] get root-owner

### PR DESCRIPTION
**Pull request by maorfr**
this command finds the top-most owner of an object (usually a Pod).
for example, the root owner of a Pod may be a ClowdApp!

### CODE REVIEW

File: tools/qontract_cli.py [hunk 1]
```diff
@@ -576,6 +577,36 @@ def quay_mirrors(ctx):
...
+                    thread_pool_size=1,
+                    settings=settings,
+                    init_api_resources=True)
+    oc = oc_map.get(cluster)
+    obj = oc.get(namespace, kind, name)
+    root_owner = oc.get_obj_root_owner(namespace, obj, allow_not_found=True,
+                                       allow_not_controller=True)
+
+    # TODO(mafriedm): fix this
+    # do not sort
```
Comment (Piojo) [inline tools/qontract_cli.py:601]: Can you clarify why you don't want this sorted?

File: tools/qontract_cli.py [hunk 2]
```diff
@@ -576,6 +577,36 @@ def quay_mirrors(ctx):
...
+    ctx.obj['options']['sort'] = False
+    # a bit hacky, but ¯\_(ツ)_/¯
```
Comment (Piojo) [inline tools/qontract_cli.py:603]: Fine by me!

File: reconcile/utils/oc.py [hunk 3]
```diff
@@ -591,21 +592,23 @@ def get_obj_root_owner(self, ns, obj, allow_not_found=False):
             ns (string): namespace of the object
             obj (dict): representation of the object
             allow_not_found (bool, optional): allow owner to be not found
+            allow_not_controller (bool, optional): allow non-controller owner
```
Comment (Piojo) [inline reconcile/utils/oc.py:595]: This would be a good place to experiment with type hints. ;)

**Review by Piojo — APPROVED** [review]
LGTM. Thanks for the test.

File: reconcile/utils/oc.py (hunk 3 above)
Comment (maorfr) [reply reconcile/utils/oc.py:595]: i'm not there yet :)

File: tools/qontract_cli.py (hunk 1 above)
Comment (maorfr) [reply tools/qontract_cli.py:601]: sorting is a bit buggy and will not work with just any object

**Pull request merged**